DM-44026: first draft of technote on wavelength-dependent PSFs (#2)
lsst-dm/dmtn-291: DMTN-291.tex

% Frequently for a technote we do not want a title page  uncomment this to remove the title page and changelog.
% use \mkshorttitle to remove the extra pages

\section{Introduction}

Rubin Observatory's massive dataset will naturally shrink the statistical errors on many important astrophysical measurements, but to take advantage of this it will need unprecedented control over systematic errors, which means careful modeling of the full system.
This is a particular challenge for aspects of the system whose behavior changes as a function of wavelength, because with broad-band filters only we cannot directly determine the spectral energy distributions (SEDs) of the objects we observe except in a very coarse sense.

> The ``zeroth-order'' challenge is photometric calibration, where a substantial fraction of the wavelength dependence comes from hardware we can inspect directly (filters, quantum efficiency, antireflective coatings, etc), and we have a comprehensive plan involving a tunable laser and the collimated beam projector to measure the wavelength-dependent throughput directly and incorporate it into our flat fielding.

Reviewer: ```suggestion
The ``zeroth-order'' challenge is photometric calibration, where a substantial fraction of the wavelength dependence comes from hardware we can inspect directly (filters, sensor quantum efficiency, antireflective coatings, etc), and we have a comprehensive plan involving a tunable laser and the collimated beam projector to measure the wavelength-dependent throughput directly and incorporate it into our flat fielding.
```

Reviewer: QE is not hardware itself but a characteristic of the sensors

The atmospheric contribution to wavelength-dependent photometric calibration will also be observed somewhat directly via the auxilliary telescope, and a sophisticated forward-modeling approach will combine all of this information with our observations of stars to yield our full photometric calibration.
This technical note is focused on higher-order wavelength dependence, which we will consider part of the point spread function (PSF), for reasons discussed in Section~\ref{sec:psfs-vs-coordinate-transforms}.

Chromatic PSF effects cannot generally be directly observed, but the most prominent one -- differential chromatic refraction (DCR) -- can be predicted accurately from the zenith angle.

> Chromatic seeing from the atmosphere \emph{roughly} scales the size of the PSF as $\lambda^{-1/5}$ \citep{1966JOSA...56.1372F,2015ApJ...807..182M}, but making use of the theory behind this in PSF modeling would require information about the structure of the atmosphere at the time of observation that we do not have.

Reviewer: Something is wrong with the citation, it renders with a '?'

Author: It looks like none of my recent additions to lsst-texmf have been picked up in this render.  I think I followed the right process, including merging my lsst-texmf branch to main and updating the submodule.

In practice, we do model the PSF as a spatially varying convolution, the coordinate transform as a general one that is only locally affine, and the photometric calibration as a spatially varying function.
But it is important to note that this decomposition is motivated by its local applicability, and hence there is a degeneracy between the spatial variation in the PSF and the coordinate transform from pixels to sky that we can resolve to our advantage: we will consider our PSF to be wavelength dependent, and the coordinate transform to to be wavelength-independent.
We believe this is the only way to resolve the degeneracy that is feasible for image processing:
\begin{itemize}
\item If the coordinate transform is wavelength-dependent, we cannot resample images to a common coordinate system to build coadds or perform image subtraction without knowing the SED of each pixel well enough to evaluate that transform and hence evaluate a resampling kernel, but we do not have any way of inferring even approximate pixel-level (as opposed to object-level) SEDs without registering images in different bands.

> \item Even if we did know the sub-band SED of each pixel contributing to a coadd prior to resampling, the result would be an effective coordinate transform that sometimes changes rapidly over just pixels (e.g. where a bright blue star's flux becomes dominant over a neighboring red star's); this cannot be a approximated as locally affine transformation, and hence our decomposition of the transfer function into PSF and coordinate transform breaks down on the coadd, and its PSF is ill-defined.

Reviewer: ```suggestion
\item Even if we did know the sub-band SED of each pixel contributing to a coadd prior to resampling, the result would be an effective coordinate transform that sometimes changes rapidly over just pixels (e.g. where a bright blue star's flux becomes dominant over a neighboring red star's); this cannot be approximated as a locally affine transformation, and hence our decomposition of the transfer function into PSF and coordinate transform breaks down on the coadd, and its PSF is ill-defined.
```

For [\ref{eqn:decomposition-final}] to hold on a coadd, we also need the coadd coordinate system to have a wavelength-independent transform from its pixel coordinates to sky coordinates, and that means the centering SED must be the same for all input observations to the coadd.
On the other hand, our decomposition of the transform is only ever a local one, and we do not necessarily need the centering SED to be the same everywhere on the coadd, and we don't even need to know what it is, as long as  all measurements correctly account for the PSF.
In practice, we expect the centering SED to be a hard-to-back-out average over the SEDs of the stars that go into our astrometric fitting.

\section{Chromatic Corrections in the Pipelines}

\subsection{Preliminary and Multi-Epoch Calibration}

The first steps of the Rubin Data Release Production (DRP) pipelines include inferring preliminary coordinate transforms and PSF models without any attention to sub-band chromaticity.
This generates lists of preliminary sources that are then spatially matched to each other and the Gaia catalog, and fed to our multi-epoch photometric and astrometric calibration routines.

> For the astrometric calibration we are focused on here, this is expected to run on large sky tiles (at least the size of the full focal plane) and include measurements all visits that overlap that tile.

Reviewer: ```suggestion
For the astrometric calibration we are focused on here, this is expected to run on large sky tiles (at least the size of the full focal plane) and include measurements of all visits that overlap that tile.
```

Gaia positions, proper motions, and parallaxes are included as a constraint in the fit (with their uncertanties), which is what sets the absolute astrometry and allows us to rigorously tie different sky tiles together.
We expect the initial DCR correction here to be sufficient to determine the final coordinate transforms for all images, despite it being non-ideal in several respects:
\begin{itemize}
  \item The SEDs used to derive the DCR correction come from colors that are themselves derived from magnitudes measured independently, and then spatially matched (rather than forced photometry).
  \item The magnitudes used to compute those colors \emph{may} only have a preliminary photometric calibration (since we are hoping to run the global photometric calibration in parallel).
  \item A catalog-space DCR correction cannot take into account how the PSF of each image interacts with the weight function used to measure the centroids that are being corrected.
\end{itemize}
If this assumption proves false, we could insert another round of astrometric fitting later in the pipeline to address both of these issues.

> This approach to DCR in astrometric calibration is the standard approach, and while we do not yet have it all running, the code we are using is at its core the same used in DECam astrometric calibrations \citep{2017PASP..129g4503B}, and we expect it to be working by DP2 (and possibly DP1, if any multi-epoch processing happens there at all).

Reviewer: ```suggestion
This approach to DCR in astrometric calibration is the standard approach, and while we do not yet have it all running, the code we are using is at its core the same used in DECam astrometric calibrations \citep{2017PASP..129g4503B}. 
We currently envisage having this working on the multi-epoch DP2 dataset. 
```

Reviewer: I would omit any reference to DP1 - if we can do it we will and it will be an added bonus (as opposed to something not delivered)

Since any PSF modeling algorithm must be able to reproduce the images of the stars it is fed at their locations and SEDs (not including noise), this should naturally provide the exact consistency we're looking for.

This approach to PSF modeling is designed to leverage the functionality of existing PSF modeling codes like Piff \citep{2021ascl.soft02024J}, which we have already integrated in the pipeline, and it should work whether the model is purely empirical (e.g. polynomial-interpolated basis functions) or more physically motivated.
Even if a full physical model is not competitive with an empirical model (they have not been, as yet), it could be advantageous to convolve the empirical model with an explicit DCR kernel, since the latter can be evaluated with no new parameters given the SED.
In any case, PSF modeling always requires careful tuning for any new survey, and sub-band wavelength dependence is very new to the field.
To our knowledge using projected positions rather than per-visit centroids has not been tried before, and it may prove difficult for unexpected reasons.
The same is true of an explicit DCR term.

> We hope to have this step working by DR1, but given the compressed commissioning timeline and the difficulty of making real progress before LSSTCa data is available, this is hard to guarantee.

Reviewer: ```suggestion
We aim to have this step working by DR1, but given the compressed commissioning timeline and the difficulty of making real progress before LSSTCam data is available, this is hard to guarantee.
```

Reviewer: "the difficulty of making real progress before LSSTCam data" -> is this due to the need for multi-epoch full-focal-plane data? or time/resources/competing priorities? If the former, suggest stating that explicitly so that it is clear that this is not a problem that can be solved by throwing more resources at it, or reprioritizing.

Reviewer: Perhaps is would be better to give DR2 as the goal and state that DR1 is a stretch goal?

Author: It's a combination of the two - I don't think we can make much progress until we get at least real ComCam data, and only some of that will carry over to LSSTCam.  And then if it was our top priority I'm pretty sure we could get it done, but it's quite likely there will be bigger problems we'll need to focus on instead.

\subsection{Coaddition}

With the PSF models and coordinate transforms finalized, we can build coadds by resampling the single-visit images to a predefined grid, and taking their weighted average (some artifact rejection also happens at this stage, but that's beyond the scope of this note).
Coadds will be built in small cells, over which the set of input visits is fixed (visits that only partially overlap a cell are not included) and the PSF can be assumed to be spatially constant.
This allows us the coadd the PSF model images with the same weights as the input visit images to form a piecewise-constant coadd PSF model.
The wavelength dependence of the PSF will be included, for example by also coadding the derivative of the PSF model with respect to one or more parameters that represent the SED.
Cell-based coaddition (including PSF coaddition) has been implemented but is still being tested prior to full integration with the main pipeline.

> We fully expect this to be the coaddition method used for DP2 (and, if coadds are built, DP1).

Reviewer: ```suggestion
We fully expect this to be the coaddition method used for DP2. 
```

Reviewer: Suggest to scrap any reference to DP1. We are only promising single-epoch data products for DP1. *if* we make coadds and they look good we can augment the DP1 dataset at the time of release.

Instead, we will provide tools to allow users to reevaluate object measurements using user-provided SEDs (though this may be much more expensive to revisit the PSF than the photometric calibration).
This is discussed further in Section~\ref{sec:sub-band-sed-inference}.

For Objects and Forced Sources, we will measure rough preliminary colors on coadds (probably as part of the deblender, which naturally runs over all bands at once) to infer SEDs.
We can then use those SEDs to evaluate the PSF wavelength dependence when performing higher-quality coadd photometry and forced photometry.
Deblending will probably not utilize the PSFs wavelength dependence.
Shear Objects (a special catalog focused on shear estimation for weak gravitational lensing) will be measured in a way that takes PSF wavelength dependence into account, but the details (including how SEDs will be inferred) are TBD.

The public Source catalog is not derived from the preliminary list of sources we matched and fed to calibration: that preliminary list only contains moderately isolated, moderately high signal-to-noise stars, and hence can be matched with little ambiguity (and ambiguous matches can be thrown away).

> The full Source catalog will be produced by redetecting and remeasuringdown to 5-sigma depth, and is considerably less clean, even though it will be built with better PSFs, backgrounds, and other calibrations.

Reviewer: ```suggestion
The full Source catalog will be produced by redetecting and remeasuring down to 5-sigma depth, and is considerably less clean, even though it will be built with better PSFs, backgrounds, and other calibrations.
```

The ``zeroth-order'' challenge is photometric calibration, where a substantial fraction of the wavelength dependence comes from hardware we can inspect directly (filters, sensor quantum efficiency, antireflective coatings, etc), and we have a comprehensive plan involving a tunable laser and the collimated beam projector to measure the wavelength-dependent throughput directly and incorporate it into our flat fielding.
The atmospheric contribution to wavelength-dependent photometric calibration will also be observed somewhat directly via the auxilliary telescope, and a sophisticated forward-modeling approach will combine all of this information with our observations of stars to yield our full photometric calibration.
This technical note is focused on higher-order wavelength dependence, which we will consider part of the point spread function (PSF), for reasons discussed in Section~\ref{sec:psfs-vs-coordinate-transforms}.

Chromatic PSF effects cannot generally be directly observed, but the most prominent one -- differential chromatic refraction (DCR) -- can be predicted accurately from the zenith angle.

> Chromatic seeing from the atmosphere \emph{roughly} scales the size of the PSF as $\lambda^{-1/5}$ \citep{1966JOSA...56.1372F,2015ApJ...807..182M}, but making use of the theory behind this in PSF modeling would require information about the structure of the atmosphere at the time of observation that we do not have.

Reviewer 2: I see "roughly" but 1/5 is Kolmogorov, and usually it's closer to 1/4;  I'd quote the range.
However, I don't see how this requires knowledge of the structure of the atmosphere.  All that knowledge (C_N^2 etc.) is encapsulated in r_0, which is directly measured and the scaling with lambda comes from the assumption about the turbulent spectrum.  Maybe that's what you mean?

Author: Rewriting the scaling as a range as suggested.

All I'm trying to say here with "structure" is that there are latent variables in a physical model we have to assume or infer from the data, and I've reworded to just say that it's a hard inference problem.

It's probably irrelevant for this note, but I'm curious what you mean by saying that `r_0` is "directly measured".  From what data?

The biggest challenge for both effects (and other smaller ones) is that they are modifications to something (i.e. the PSF) that is already hard to infer from the data, and to include them in that inference we need to know the SEDs of the stars that go into the PSF model.
And like the photometric calibration, we also need to infer the SEDs of other objects in order to use the PSF to measure them.

\section{PSFs vs. Coordinate Transforms}

\label{sec:psfs-vs-coordinate-transforms}

The value $z$ at some position $\mathbf{x}$ on a detector is related to the true flux $f$ (a function of sky position $\mathbf{r}$ and wavelength $\lambda$) by some transfer function $T$:

\begin{equation}
  z(\mathbf{x}) = \int\!\! d^2 \mathbf{r} \int\!\! d\!\lambda
    \, T(\mathbf{x},\mathbf{r},\lambda)
    \, f(\mathbf{r},\lambda)
  \label{eqn:transfer-fn}
\end{equation}

> 

Reviewer 2: Isn't this a convolution?  And probably not an issue for this audience as the discussion of seeing is tangential, but $T$ usually means the OTF, the Fourier transform of the PSF, in that community.

Author: It's a convolution only if the difference of `x` and `r` is all that affects `T`.  This is a general function of both sky position and pixel position, so it can represent spatially-varying convolutions, coordinate transforms, etc. as well.

I'll switch from `T` to `\Psi` throughout.

The normalization of $T$ is the photometric calibration, which we'll call $\alpha$.

> Its dependence on $\mathbf{r}$ and $\mathbf{x}$ encodes both the PSF $\phi$ and the coordinate transform $\mathbf{S}$ from sky to pixel coordinates (frequently called the WCS, or ``world coordinate system''), i.e. we approximate [\ref{eqn:transfer-fn}] \emph{locally} as either

Reviewer 2: Will using the acronym WCS here trigger e.g. Dave Monet?  What you say is clear

Author: I figure it's still a useful reference as a way to clarify what I'm talking about.  I think the assumption that WCS == FITS WCS is rarely made these days.

\begin{equation}
  z(\mathbf{x}) =  \int\!\! d^2 \mathbf{r} \int\!\! d\!\lambda
    \; \alpha(\lambda)
    \; \phi_r\!\left(\mathbf{S}_\lambda(\mathbf{x}) - \mathbf{r}, \lambda\right)
    \; f\!\left(\mathbf{r}, \lambda\right)
\end{equation}
with the PSF defined in sky coordinates, or
\begin{equation}
  z(\mathbf{x}) =  \int\!\! d^2 \mathbf{x} \int\!\! d\!\lambda
    \;\alpha(\lambda)
    \; \phi_x\!\left(\mathbf{x} - \mathbf{S}_\lambda^{-1}(\mathbf{r}), \lambda\right)
    \; f\!\left(S^{-1}_\lambda(\mathbf{r}), \lambda\right)
    \; \left|\frac{d \mathbf{S}}{d\mathbf{r}}\right|^{-1}
\end{equation}
in pixel coordinates.

> Note that there is no spatial variation in the PSF or the photometric calibration in either case - that is the first reason this is a local approximation only.

Reviewer 2: Sorry, more notation.   Željko will object that he's written photometric calibration white papers which use S as the throughput as a function of frequency (I think; certainly a photometric symbol).  He has a point, as you'll be looking at SEDs too

Author: I'll switch from `S` to `A` throughout.

These are are both valid if the transform $S$ is approximately locally affine (e.g. on the scale of the PSF); when it is not, neither is necessarily correct -- one may be somewhat better than the other, depending on the details of $T$, but when this approximation breaks down it is safest to consider the data unusable.

> Thankfully this second local approximation is almost always valid, with small regions affected by strong lateral electric field differences in the detectors a notable exception, e.g. the ``tape bumps'' seen in the Dark Energy Camera (DECam).

Reviewer 2: or small-scale tree-rings, or "epoxy blobs" in HSC, which we don't correct for.  Incidentally, I'm about to start correcting for them in PFS

Author: I've added another sentence noting that tree rings are at a scale that is worrying, but that current corrections for those still make this assumption and appear to get away with it.

Lateral electric field distortions caused by charge from sources (i.e. the ``brighter fatter'' effect) do not fit in this decomposition at all, which is why we correct this in the images as early as possible rather than include it in (say) the PSF model for downstream processing.

In practice, we do model the PSF as a spatially varying convolution, the coordinate transform as a general one that is only locally affine, and the photometric calibration as a spatially varying function.

> But it is important to note that this decomposition is motivated by its local applicability, and hence there is a degeneracy between the spatial variation in the PSF and the coordinate transform from pixels to sky that we can resolve to our advantage: we will consider our PSF to be wavelength dependent, and the coordinate transform to to be wavelength-independent.

Reviewer 2: And the measured centre of the PSF, not just the variation thereof -- we could offset the PSF model and the astrometry and it wouldn't change the science. 

I'd have written "centroid" until a few weeks ago, when I realised that the diffraction spikes (or worse optical LSF in a spectrograph) don't have a well defined centroid.  I don't think this matters in practice!

Author: Agreed, but I think at this stage in the note bringing this up would be more of a distraction; it's a point I think I'm aleady making later.

We believe this is the only way to resolve the degeneracy that is feasible for image processing:
\begin{itemize}

> \item If the coordinate transform is wavelength-dependent, we cannot resample images to a common coordinate system to build coadds or perform image subtraction without knowing the SED of each pixel well enough to evaluate that transform and hence evaluate a resampling kernel, but we do not have any way of inferring even approximate pixel-level (as opposed to object-level) SEDs without registering images in different bands.

Reviewer 2: I don't think that a "pixel SED" is well defined.  It would be for a Kaiser coadd, where we're estimating the deconvolved sky and then reconvolving with a chosen PSF;  but then the choice of that PSF is what you're discussing.
I'm not sure if this is worth elucidating here.

Author: Agreed; for "pixel SED" we need to state what it is that a pixel represents in terms of some likelihood, and I think that's well defined for Kaiser coadds and not obviously defined otherwise.  But I think I'd rather not get into that here.

> \item Even if we did know the sub-band SED of each pixel contributing to a coadd prior to resampling, the result would be an effective coordinate transform that sometimes changes rapidly over just pixels (e.g. where a bright blue star's flux becomes dominant over a neighboring red star's); this cannot be approximated as a locally affine transformation, and hence our decomposition of the transfer function into PSF and coordinate transform breaks down on the coadd, and its PSF is ill-defined.

Reviewer 2: I'm not sure if this is right.  If you had sub-band SEDs for each pixel (see previous comment!) you'd have sliced the image into multiple images, each with a well-defined transformation.  That would handle the blue-and-red-light in a pixel problem.

I'm not saying that I disagree with the conclusion.

Author: Yes, I've thought about that kind of Kaiser coadd and even did a little math on it.  At best the coadds for different sub-bands have really strongly correlated values, and unless you have so many sub-bands that you Nyquist-sample the spectral-domain PSF, the correlation structure is really messy and you can't use FFTs to solve it.

But I think again that's probably a distraction for anyone else reading this note.

But for the most part, we prefer to have one convention for how to decompose the wavelength-dependent transfer function, and a wavelength-dependent spatially varying PSF and achromatic coordinate transform are the clear choice as this is valid throughout the full pipeline.

For the rest of this note, we will thus drop the wavelength dependence on $S$, in addition to representing its local affine version in augmented matrix form, i.e.:
\begin{equation}
  \mathbf{r} = \mathbf{S} \mathbf{x}
\end{equation}
\begin{equation}
\begin{bmatrix}
    r_0 \\
    r_1 \\
    1
  \end{bmatrix} =
  \begin{bmatrix}
    S_{0, 0} & S_{0, 1} & S_0 \\
    S_{1, 0} & S_{1, 1} & S_1 \\
    0 & 0 & 1
  \end{bmatrix}
  \begin{bmatrix}
    x_0 \\
    x_1 \\
    1
  \end{bmatrix}
\end{equation}
With the PSF in sky coordinates (as we'll prefer for notational simplicity only), we then have:
\begin{equation}
  z(\mathbf{x}) =  \int\!\! d^2 \mathbf{r} \int\!\! d\!\lambda
    \; \alpha_r(\mathbf{r}, \lambda)
    \; \phi_r\!\left(\mathbf{S}\mathbf{x} - \mathbf{r}, \lambda\right)
    \; f\!\left(\mathbf{r}, \lambda\right)
  \label{eqn:decomposition-final}
\end{equation}

> A major implication of this convention is that the PSF model cannot not be simultaneously centered (i.e. have zero first moment under a certain weight function) for all wavelengths when a non-isotropic effect like DCR is in play.

Reviewer 2: I think that a weighted first-order moment is a sensible definition, but it's actually only _optimal_ for a Gaussian PSF.  In that limit -- and I think _only_ in that limit -- adaptive moments provide a way to estimate that weighting function.   In general we will have to specify this externally to the centroiding algorithm, or accept some loss of efficiency (which is probably fine).

Again, I don't think that this changes any of your conclusions

Author: I'll fix the parenthetical to say that the PSF cannot be centered at all wavelengths for any useful definition of "centered", which I think is more general and still true (though "useful" may be doing some heavy lifting).

Instead, evaluating the PSF model for an observation at the true coordinates of a star will yield an image that is offset (and smeared) such that it matches where the star actually appears in that observation.
On a coadd constructed with many observations with the zenith direction rotated, the effective coadd PSF will be the weighted sum of those directionally-smeared single-epoch PSFs \citep{2023OJAp....6E...5M}, and hence be spread out in all directions (in the limit of many zenith direction rotations).

> A corollary of this is that the PSF model \emph{is} centered for \emph{some} SED, which we will refer to as the ``centering SED``.

Reviewer 2: For some SED and some weighting function, which may or may not be chromatic.   If we choose a chromatic weight function, it isn't actually obvious to me that there is only one SED which results in a centered PSF.   Actually, given weird enough SEDs, I bet that there are a range (e.g. two delta functions at suitable wavelengths)

Author: I certainly hadn't considered the possibility of using a chromatic weight function.  I don't think it'd help with anything, and it's make for a chicken-and-egg-problem.  But I suppose I should be explicit that I'm assuming it's achromatic.

I agree that saying that the PSF model is centered for one SED is an oversimplification, and I'll weaken that language.  It's an integral constraint.

For [\ref{eqn:decomposition-final}] to hold on a coadd, we also need the coadd coordinate system to have a wavelength-independent transform from its pixel coordinates to sky coordinates, and that means the centering SED must be the same for all input observations to the coadd.
On the other hand, our decomposition of the transform is only ever a local one, and we do not necessarily need the centering SED to be the same everywhere on the coadd, and we don't even need to know what it is, as long as  all measurements correctly account for the PSF.

> In practice, we expect the centering SED to be a hard-to-back-out average over the SEDs of the stars that go into our astrometric fitting.

Reviewer 2: I think that the choice of weight functions probably _does_ matter here.  It's not the same for every epoch, so the "centering SED" probably isn't either.  Do we explicitly offset the PSF models to fix this?

Author: I think my hope was that by fitting the astrometry and PSF as I proposed in later sections, the "centering SED" would end up corresponding to some local average of the SEDs of the various stars that contribute to the astrometric fit, weighted by how they contributed to the local result of the fit and PSF model.  And then the "centering SED" would end up being roughly the same for every epoch, since the stars that go into the astrometric fit should be bright enough that they appear on almost every epoch are are selected jointly across all epochs.  There are a lot of assumptions there, but I don't have an alternative in mind, and I think they're reasonable assumptions for the kind of first-order correction we need to do.

\section{Chromatic Corrections in the Pipelines}

\subsection{Preliminary and Multi-Epoch Calibration}

The first steps of the Rubin Data Release Production (DRP) pipelines include inferring preliminary coordinate transforms and PSF models without any attention to sub-band chromaticity.

> This generates lists of preliminary sources that are then spatially matched to each other and the Gaia catalog, and fed to our multi-epoch photometric and astrometric calibration routines.

Reviewer 2: Are we requiring that all these "sources" are stars?  I think so; astrometry with galaxies is hard (hence my interest in diffim kernels)

Author: Yes, and I think chromatic astrometry with galaxies is even harder, because their SEDs are hard to infer, and I don't think difference-kernel games help with that (at least not as long as those kernels are achromatic).

For the astrometric calibration we are focused on here, this is expected to run on large sky tiles (at least the size of the full focal plane) and include measurements of all visits that overlap that tile.

> The coordinate transform model will include static or nearly static optical and sensor-level distortion models that are fit in advance, composed with pointing, rotation, and atmospheric turbulence terms for each visit.

Reviewer 2: I'd say, "optical and camera" as the CCDs will probably move

The true positions, proper motions, and parallaxes of the input stars are inferred during this fit as well, and here we expect to include a catalog-space preliminary DCR correction as well.
Gaia positions, proper motions, and parallaxes are included as a constraint in the fit (with their uncertanties), which is what sets the absolute astrometry and allows us to rigorously tie different sky tiles together.
We expect the initial DCR correction here to be sufficient to determine the final coordinate transforms for all images, despite it being non-ideal in several respects:
\begin{itemize}

>   \item The SEDs used to derive the DCR correction come from colors that are themselves derived from magnitudes measured independently, and then spatially matched (rather than forced photometry).

Reviewer 2: Clarify that we are not using GAIA colours here?

  \item The magnitudes used to compute those colors \emph{may} only have a preliminary photometric calibration (since we are hoping to run the global photometric calibration in parallel).
  \item A catalog-space DCR correction cannot take into account how the PSF of each image interacts with the weight function used to measure the centroids that are being corrected.
\end{itemize}
If this assumption proves false, we could insert another round of astrometric fitting later in the pipeline to address both of these issues.

This approach to DCR in astrometric calibration is the standard approach, and while we do not yet have it all running, the code we are using is at its core the same used in DECam astrometric calibrations \citep{2017PASP..129g4503B}. 
We currently envisage having this working on the multi-epoch DP2 dataset. 

\subsection{Final PSF Modeling}

After the astrometric calibration is complete, we will re-fit the PSF model for every visit separately.

> The full parameterization of this model is TBD, including how wavelength dependence will be represented, but prior art (from the Dark Energy Survey collaboration, still in prep) suggests a simple regression against a broad-band color may be sufficient.

Reviewer 2: Add "such as $g - i$" ?

The stars used in the PSF model will be selected from the same set of matched preliminary measurements used in the astrometric and photometric calibrations, and hence will have the same inferred SEDs available.

> The linchpin of our plan to ensure consistency between the PSF model and coordinate transform is that the PSF modeling will be fed star positions derived by projecting the best-fit position for each star from the astrometric fit (also incorporating proper motion and parallax at the epoch of the observation, of course), instead of per-epoch centroids.

Reviewer 2: "also incorporating" -> "after correcting for"?

Since any PSF modeling algorithm must be able to reproduce the images of the stars it is fed at their locations and SEDs (not including noise), this should naturally provide the exact consistency we're looking for.

This approach to PSF modeling is designed to leverage the functionality of existing PSF modeling codes like Piff \citep{2021ascl.soft02024J}, which we have already integrated in the pipeline, and it should work whether the model is purely empirical (e.g. polynomial-interpolated basis functions) or more physically motivated.
Even if a full physical model is not competitive with an empirical model (they have not been, as yet), it could be advantageous to convolve the empirical model with an explicit DCR kernel, since the latter can be evaluated with no new parameters given the SED.
In any case, PSF modeling always requires careful tuning for any new survey, and sub-band wavelength dependence is very new to the field.

> To our knowledge using projected positions rather than per-visit centroids has not been tried before, and it may prove difficult for unexpected reasons.

Reviewer 2: I was probably skimming too fast, but what do you mean by "projected positions"?  Model positions?

Author: Yes, model positions.  Will use that instead.

The same is true of an explicit DCR term.
We hope to have this step working by DR1, but given the compressed commissioning timeline and the difficulty of making real progress before LSSTCa data is available, this is hard to guarantee.

\subsection{Coaddition}

With the PSF models and coordinate transforms finalized, we can build coadds by resampling the single-visit images to a predefined grid, and taking their weighted average (some artifact rejection also happens at this stage, but that's beyond the scope of this note).

> Coadds will be built in small cells, over which the set of input visits is fixed (visits that only partially overlap a cell are not included) and the PSF can be assumed to be spatially constant.

Reviewer 2: I'd explain "the PSF can be assumed to be spatially constant".  Say that we plan to make the cells small enough that the PSF is essentially constant, and that this assumption is not fundamental?

This allows us the coadd the PSF model images with the same weights as the input visit images to form a piecewise-constant coadd PSF model.

> The wavelength dependence of the PSF will be included, for example by also coadding the derivative of the PSF model with respect to one or more parameters that represent the SED.

Reviewer 2: I don't know what this means.  We don't have per-source SEDs (as you point out, more than one source can lie within a pixel), so aren't we using a single PSF per epoch?  Do you mean we'll choose the SED of the PSF to be e.g. flat in $\nu F_\nu$ or something different?

Author: I'll rewrite this to say that we'll expand the wavelength dependence of the PSF at each point in some basis and coadd the basis expansion.  It'll probably be convenient if it's the sort of basis where there's a zeroth-order term that corresonds to a flat spectrum or something similarly convenient, but it doesn't have to be for this scheme to work.

> Cell-based coaddition (including PSF coaddition) has been implemented but is still being tested prior to full integration with the main pipeline.

Reviewer 2: Do we want to say that we are exploring this in conjunction with DES, or that the Y6 DES release is using the same approach (is it?).   Something to make cell-based coadds less scary to those outside Rubin?

We fully expect this to be the coaddition method used for DP2. 

\subsection{Inferring SEDs}

In order to use a chromatic PSF model (or a chromatic photometric calibration), we need to infer the SEDs of the objects we're measuring, and unlike previous stages this now includes galaxies, QSOs, supernovae, and other astrophysical objects whose SEDs are much more complex than stars.
Our goal for SED inference is to have a simple, easily-reproducible mapping from broadband colors to SED, and to attempt to transition smoothly from something extremely simple (e.g. splines, step functions) for most objects to standard stellar SED models for objects that have a high probability of being a star.

> In particular, we do not currently plan to perform a full photometric redshift analysis to infer the SEDs of galaxies prior to performed PSF-based measurements on them; not only would this require at least two passes of photometric redshift estimation (it can't be done well unless PSF-corrected photometry is used as an \emph{input}), Rubin Data Management simply isn't scoped to do sophisticated photometric redshifts.

Reviewer 2: performed -> performing ?

> In particular, we do not currently plan to perform a full photometric redshift analysis to infer the SEDs of galaxies prior to performed PSF-based measurements on them; not only would this require at least two passes of photometric redshift estimation (it can't be done well unless PSF-corrected photometry is used as an \emph{input}), Rubin Data Management simply isn't scoped to do sophisticated photometric redshifts.

Reviewer 2: Do we want to mention that Rubin DM _is_ expecting to put out some photo-z estimates?  It's tangential to this note, but I wouldn't want to scare people.

Author: I'll just drop the second argument about not wanting to do sophisticated photo-zs; I don't think it adds much.

> In particular, we do not currently plan to perform a full photometric redshift analysis to infer the SEDs of galaxies prior to performed PSF-based measurements on them; not only would this require at least two passes of photometric redshift estimation (it can't be done well unless PSF-corrected photometry is used as an \emph{input}), Rubin Data Management simply isn't scoped to do sophisticated photometric redshifts.

Reviewer 2: We don't actually need "sophisticated photo-z"s;  all we need is the SED given the colours, and a 4000\AA and a Lyman break are the same for us.   But I think your point remains

Instead, we will provide tools to allow users to reevaluate object measurements using user-provided SEDs (though this may be much more expensive to revisit the PSF than the photometric calibration).
This is discussed further in Section~\ref{sec:sub-band-sed-inference}.

For Objects and Forced Sources, we will measure rough preliminary colors on coadds (probably as part of the deblender, which naturally runs over all bands at once) to infer SEDs.
We can then use those SEDs to evaluate the PSF wavelength dependence when performing higher-quality coadd photometry and forced photometry.
Deblending will probably not utilize the PSFs wavelength dependence.
Shear Objects (a special catalog focused on shear estimation for weak gravitational lensing) will be measured in a way that takes PSF wavelength dependence into account, but the details (including how SEDs will be inferred) are TBD.

The public Source catalog is not derived from the preliminary list of sources we matched and fed to calibration: that preliminary list only contains moderately isolated, moderately high signal-to-noise stars, and hence can be matched with little ambiguity (and ambiguous matches can be thrown away).

> The full Source catalog will be produced by redetecting and remeasuring down to 5-sigma depth, and is considerably less clean, even though it will be built with better PSFs, backgrounds, and other calibrations.

Reviewer 2: "to 5-sigma depth"  on the coadds?  And "less clean" is a statement about the sky, not the processing

Author: I'll say that matching them would be less clean due to increased blending.

\subsection{Difference Image Analysis}

% Ian and Eric need to review this section: I have no idea how out-of-date my
% understanding of the situation is, or even if it was ever all that accurate.

Image subtraction in the presence of chromatic PSFs needs more than just chromatic PSF models: it also requires \emph{pixel}-level SED inference, because the PSF matching kernel cannot be a function of wavelength, and the inclusion of that SED information in the template.
This represents a computationally challenging inference problem to brute-force, but Rubin DM has developed an efficient algorithm with a few simplifying assumptions \citep{DMTN-037, DMTN-121} that has performed reasonably well on simple simulations.
The bigger problem, at least early in the survey, is that we simply don't expect to have enough visits at different airmasses over enough of the sky to constrain the per-pixel SEDs.
As a result, we expect to perform image subtraction without correcting for DCR or other chromatic effects until we are a few years into the survey, and will revisit these algorithms then.

> In the meantime, we expect machine learning-based reliability algorithms to do a reasonably good job of rejecting DCR-induced dipoles.

Reviewer 2: Dipoles simply don't seem to have been the problem that people expected, and I don't know why.   If they _are_ a problem, I'd expect to be able to handle them without resorting to ML (or, to put it another way, to be able to make the ML's job easier by cleaning the samples);  whether that's necessary remains to be seen.

> \section{Sub-band SED Inference from PSFs}

Reviewer 2: I'm going to discuss this in person with Jim, so I'll submit the other $\pm$ picky comments now
